ui/src/criteria/attribute.tsx

```diff
@@ -0,0 +1,131 @@
+import { Checkbox, FormControlLabel, ListItem } from "@mui/material";
+import Stack from "@mui/material/Stack";
+import Typography from "@mui/material/Typography";
+import { CriteriaConfig, CriteriaPlugin, registerCriteriaPlugin } from "cohort";
+import produce from "immer";
+import React from "react";
+import * as tanagra from "tanagra-api";
+import { useUnderlay } from "../hooks";
+
+type Selection = {
+  id: number;
+  name: string;
+};
+
+interface Config extends CriteriaConfig {
+  entity: string;
+}
+
+interface Data extends Config {
+  selected: Selection[];
+}
+
+type AttributeEditProps = {
+  dispatchFn: (data: Data) => void;
+  data: Data;
+};
+
+@registerCriteriaPlugin("attribute", (config: CriteriaConfig) => ({
+  ...(config.plugin as Config),
+  selected: [],
+}))
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+class _ implements CriteriaPlugin<Data> {
+  public data: Data;
+
+  constructor(public id: string, data: unknown) {
+    this.data = data as Data;
+  }
+
+  renderEdit(dispatchFn: (data: Data) => void) {
+    return <AttributeEdit dispatchFn={dispatchFn} data={this.data} />;
+  }
+
+  renderDetails() {
+    return <ConceptDetails data={this.data} />;
+  }
+
+  generateFilter(entityVar: string, fromOccurrence: boolean) {
+    if (this.data.selected.length === 0) {
+      return null;
+    }
+    console.log(entityVar);
+    console.log(fromOccurrence);
+    return null;
+  }
+
+  occurrenceEntities() {
+    return [];
+  }
+}
+
+function AttributeEdit(props: AttributeEditProps) {
+  const underlay = useUnderlay();
+  const attributeName = props.data.entity;
```

Comment:
This can't be correct. It's either an entity or an attribute. From the usage it looks like it's an attribute so change it to `attribute` in `Data`.

Reply:
got it. change the data structure in App.tsx file from "entity" to "attribute". and also change the following parameter in attribute.tsx